src/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Windows.cs

```diff
@@ -1082,63 +1079,50 @@ internal static string GetProcessShortName(String name)
 
         // native struct defined in ntexapi.h
         [StructLayout(LayoutKind.Sequential)]
-        internal class SystemProcessInformation
+        internal unsafe struct SystemProcessInformation
         {
             internal uint NextEntryOffset;
             internal uint NumberOfThreads;
-            private long _SpareLi1;
-            private long _SpareLi2;
-            private long _SpareLi3;
-            private long _CreateTime;
-            private long _UserTime;
-            private long _KernelTime;
+
+            private fixed byte reserved1[48];
 
             internal ushort NameLength;   // UNICODE_STRING   
             internal ushort MaximumNameLength;
             internal IntPtr NamePtr;     // This will point into the data block returned by NtQuerySystemInformation
 
             internal int BasePriority;
             internal IntPtr UniqueProcessId;
-            internal IntPtr InheritedFromUniqueProcessId;
+            private IntPtr reserved2;
             internal uint HandleCount;
             internal uint SessionId;
-            internal UIntPtr PageDirectoryBase;
+            private UIntPtr reserved3;
             internal UIntPtr PeakVirtualSize;  // SIZE_T
             internal UIntPtr VirtualSize;
-            internal uint PageFaultCount;
+            private uint reserved4;
 
             internal UIntPtr PeakWorkingSetSize;
             internal UIntPtr WorkingSetSize;
-            internal UIntPtr QuotaPeakPagedPoolUsage;
+            private UIntPtr reserved5;
             internal UIntPtr QuotaPagedPoolUsage;
-            internal UIntPtr QuotaPeakNonPagedPoolUsage;
+            private UIntPtr reserved6;
             internal UIntPtr QuotaNonPagedPoolUsage;
             internal UIntPtr PagefileUsage;
             internal UIntPtr PeakPagefileUsage;
             internal UIntPtr PrivatePageCount;
 
-            private long _ReadOperationCount;
-            private long _WriteOperationCount;
-            private long _OtherOperationCount;
-            private long _ReadTransferCount;
-            private long _WriteTransferCount;
-            private long _OtherTransferCount;
+            private fixed byte reserved7[48];
         }
 
         [StructLayout(LayoutKind.Sequential)]
-        internal class SystemThreadInformation
+        internal unsafe struct SystemThreadInformation
         {
-            private long _KernelTime;
-            private long _UserTime;
-            private long _CreateTime;
-
-            private uint _WaitTime;
+            private fixed byte reserved1[28];
```

Comment:
> Because in the header they are coalesced

Those can be done using fixed, though, as part of closely matching what Windows has.  And I don't understand why layout is any more challenging when using fixed rather than writing out each subtype N times: they end up doing the same thing, but one requires more code, doesn't closely match the Windows header, and requires making up names that then force further differences from the Windows header.  What am I missing?